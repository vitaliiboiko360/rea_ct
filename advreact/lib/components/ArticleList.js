import React from 'react';

import ArticleContainer from './Article';

const ArticleList = (props) => {
  return (
    <div>
      {Object.values(props.articles).map(article =>
        <ArticleContainer
          key={article.id}
          article={article}
        />
      )}
    </div>
  );
};

export default ArticleList;
