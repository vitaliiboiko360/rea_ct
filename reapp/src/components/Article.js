import React from 'react';

const styles = {
    article: {
    paddingBottom: 5,
    marginBottom: 5,
    }
}

const Article = (props) => {
    const {article, author} = props;
    return (
        <div>
            <div style={styles.article} >{article.title}</div>
            <div>{article.date}</div>
            <div>
                <a href={author.website}>
                    {author.firstName} {author.secondName}
                </a>
            </div>
            <div>{article.body}</div>
        </div>
    );
};

export default Article;