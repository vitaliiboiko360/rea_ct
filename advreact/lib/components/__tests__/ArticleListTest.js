import React from 'react';
import ArticleList from '../ArticleList';
import Article from '../Article'

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

import { shallow } from 'enzyme';

describe('ArticleList', () => {

  const testProps = {
    articles: {
      a: { id: 'a' },
      b: { id: 'b' },
    },
  };

  Article.propTypes = {};

  it('renders correctly', () => {
    const wrapper = shallow(
      <ArticleList
        {...testProps}
      />
    );

    expect(wrapper.find('ArticleContainer').length).toBe(2);

    expect(wrapper.getElement()).toMatchSnapshot();
  });

});
