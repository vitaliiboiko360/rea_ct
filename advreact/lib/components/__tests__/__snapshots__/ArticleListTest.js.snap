// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList renders correctly 1`] = `
<div>
  <Article
    article={
      Object {
        "id": "a",
      }
    }
  />
  <Article
    article={
      Object {
        "id": "b",
      }
    }
  />
</div>
`;
